### PR TITLE
Add `request` and `options` to `HTTPError`

### LINKED ISSUE #287: Include request object in HTTPError
I would suggest attaching request object to `HTTPError`. It would be useful to log request body, credentials policy etc. For now I have to use a custom http error, which leads to some problems with retry logic, so now I have to write a `beforeRetry` hook that checks whether request should be retried or not. It doesn't seem to do any damage to include request in `HTTPError`, so why not to do it? Also you could include `options` as well, because after normalization they could be used for logging as well

### PR DESCRIPTION
**Pull request by ialexi-bl**
Fixes #287

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -414,8 +414,10 @@ export interface ResponsePromise extends Promise<Response> {
 The error has a response property with the `Response` object.
 */
 declare class HTTPError extends Error {
-	constructor(response: Response);
+	constructor(response: Response, request: Request, options: NormalizedOptions);
 	response: Response;
+	request: Request;;
+	options: NormalizedOptions;;
```
Comment (sindresorhus) [inline index.d.ts:420]: Typo

File: index.test-d.ts [hunk 2]
```diff
@@ -23,7 +23,13 @@ for (const method of requestMethods) {
 
 expectType<typeof ky>(ky.create({}));
 expectType<typeof ky>(ky.extend({}));
-expectType<InstanceType<typeof ky.HTTPError>>(new ky.HTTPError(new Response()));
+expectType<InstanceType<typeof ky.HTTPError>>(
+	new ky.HTTPError(
+		new Response(),
+		new Request("http://example.com"),
```
Comment (sindresorhus) [inline index.test-d.ts:29]:
```suggestion
		new Request('https://example.com'),
```

**Comment by sindresorhus** [comment]
You also need to update the TS doc comment and the readme.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js [hunk 3]
```diff
@@ -314,7 +316,7 @@ class Ky {
 			this._decorateResponse(response);
 
 			if (!response.ok && this._options.throwHttpErrors) {
-				throw new HTTPError(response);
+				throw new HTTPError(response, this.request, this._optoins);
```
Comment (dan-lee) [inline index.js:319]:
```suggestion
				throw new HTTPError(response, this.request, this. _options);
```

Comment (nielsdB97) [reply index.js:319]: Watch out, there's an added space there ☝️ 

```suggestion
				throw new HTTPError(response, this.request, this._options);
```

Comment (dan-lee) [reply index.js:319]: Thanks for spotting @nielsdB97 👀!  I edited the suggestion

**Comment by ialexi-bl** [comment]
Why is it displaying "Changes requested"? I've added all of the suggestions in the last commit it seems

File: index.d.ts [hunk 4]
```diff
@@ -411,13 +411,13 @@ export interface ResponsePromise extends Promise<Response> {
 }
 
 /**
-The error has a response property with the `Response` object.
+The error has a response, request and options properties that contain `Response`, `Request` and `NormalizedOptions` objects respectively
```
Comment (sindresorhus) [inline index.d.ts:414]: Should end with a dot. And this text and the readme docs should be in sync.

**Comment by sholladay** [comment]
`index.d.ts` needs to be synced with the readme, as Sindre mentioned, but otherwise LGTM.

**Pull request merged**